app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/firestore/FirestoreDatabaseService.kt

```diff
@@ -70,11 +71,36 @@ class FirestoreDatabaseService(
         }
     }
 
+    override suspend fun getAllPictures(category: Category): Set<CategorizedPicture> {
+        TODO("Not yet implemented")
+    }
+
+    override suspend fun removeCategory(category: Category) {
+        TODO("Not yet implemented")
+    }
+
+    override suspend fun removePicture(picture: CategorizedPicture) {
+        TODO("Not yet implemented")
+    }
+
     override suspend fun putDataset(name: String, categories: Set<Category>): Dataset {
         TODO("Not yet implemented")
     }
 
-    override suspend fun getDataset(name: String): Dataset? {
+    override suspend fun getDataset(id: Any): Dataset? {
```

Comment:
Yes because it allows to be more flexible and since the ids are never provided by the "users"  ids used will always be of the type expected by the implementation.